lib/new_relic/agent/instrumentation/concurrent_ruby.rb

```diff
@@ -0,0 +1,30 @@
+# This file is distributed under New Relic's license terms.
+# See https://github.com/newrelic/newrelic-ruby-agent/blob/main/LICENSE for complete details.
+# frozen_string_literal: true
+
+require_relative 'concurrent_ruby/instrumentation'
+require_relative 'concurrent_ruby/chain'
+require_relative 'concurrent_ruby/prepend'
+
+DependencyDetection.defer do
+  named :'concurrent_ruby'
+
+  depends_on do
+    defined?(Concurrent)
+  end
+
+  executes do
+    NewRelic::Agent.logger.info('Installing concurrent-ruby instrumentation')
+
+    if use_prepend?
+      prepend_instrument(Concurrent::ThreadPoolExecutor, NewRelic::Agent::Instrumentation::ConcurrentRuby::Prepend)
+
+      [Concurrent::Promises.const_get(:'InternalStates')::Rejected,
+        Concurrent::Promises.const_get(:'InternalStates')::PartiallyRejected].each do |klass|
+          klass.prepend(NewRelic::Agent::Instrumentation::ConcurrentRuby::ErrorPrepend)
```

Comment:
After chatting with @tannalynn, `klass.prepend` is used instead of `prepend_instrument` because all `prepend_instrument` adds is logging around the instrumented library. Since the logging occurs on line 20, we don't need to repeat it here.